Non-deterministic tensorflow test failing

This test is failing:
org.apache.wayang.tensorflow.model.TensorflowModelTest with the following message:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0:test (default-test) on project wayang-tensorflow: 
	
[ERROR] 
	
[ERROR] Please refer to /home/jenkins/jenkins-agent/workspace/Wayang_Wayang_main/wayang-platforms/wayang-tensorflow/target/surefire-reports for the individual test results.
	
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
	
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
	
[ERROR] Command was /bin/sh -c cd '/home/jenkins/jenkins-agent/workspace/Wayang_Wayang_main/wayang-platforms/wayang-tensorflow' && '/usr/local/asfpackages/java/adoptium-jdk-17.0.12+7/bin/java' '--add-exports=java.base/sun.nio.ch=ALL-UNNAMED' '--add-exports=java.base/jdk.internal.misc=ALL-UNNAMED' '--add-opens=java.base/java.nio=ALL-UNNAMED' '--add-opens=java.base/sun.nio.ch=ALL-UNNAMED' '--add-opens=java.base/java.lang=ALL-UNNAMED' '--add-opens=java.base/java.util=ALL-UNNAMED' '--add-opens=java.base/java.io=ALL-UNNAMED' '--add-opens=java.base/java.lang.reflect=ALL-UNNAMED' '--add-opens=java.base/sun.reflect.annotation=ALL-UNNAMED' '--add-opens=java.base/java.util.concurrent=ALL-UNNAMED' '--add-opens=java.base/java.net=ALL-UNNAMED' '--add-opens=java.base/java.lang.invoke=ALL-UNNAMED' 'org.apache.maven.surefire.booter.ForkedBooter' '/home/jenkins/jenkins-agent/workspace/Wayang_Wayang_main/wayang-platforms/wayang-tensorflow/target/surefire' '2026-06-29T08-37-50_461-jvmRun1' 'surefire-20260629084402701_84tmp' 'surefire_8-20260629084402701_85tmp'
	
[ERROR] Error occurred in starting fork, check output in log
	
[ERROR] Process Exit Code: 134
	
[ERROR] Crashed tests:
	
[ERROR] org.apache.wayang.tensorflow.model.TensorflowModelTest
	
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
	
[ERROR] Command was /bin/sh -c cd '/home/jenkins/jenkins-agent/workspace/Wayang_Wayang_main/wayang-platforms/wayang-tensorflow' && '/usr/local/asfpackages/java/adoptium-jdk-17.0.12+7/bin/java' '--add-exports=java.base/sun.nio.ch=ALL-UNNAMED' '--add-exports=java.base/jdk.internal.misc=ALL-UNNAMED' '--add-opens=java.base/java.nio=ALL-UNNAMED' '--add-opens=java.base/sun.nio.ch=ALL-UNNAMED' '--add-opens=java.base/java.lang=ALL-UNNAMED' '--add-opens=java.base/java.util=ALL-UNNAMED' '--add-opens=java.base/java.io=ALL-UNNAMED' '--add-opens=java.base/java.lang.reflect=ALL-UNNAMED' '--add-opens=java.base/sun.reflect.annotation=ALL-UNNAMED' '--add-opens=java.base/java.util.concurrent=ALL-UNNAMED' '--add-opens=java.base/java.net=ALL-UNNAMED' '--add-opens=java.base/java.lang.invoke=ALL-UNNAMED' 'org.apache.maven.surefire.booter.ForkedBooter' '/home/jenkins/jenkins-agent/workspace/Wayang_Wayang_main/wayang-platforms/wayang-tensorflow/target/surefire' '2026-06-29T08-37-50_461-jvmRun1' 'surefire-20260629084402701_84tmp' 'surefire_8-20260629084402701_85tmp'
	
[ERROR] Error occurred in starting fork, check output in log
	
[ERROR] Process Exit Code: 134
	
[ERROR] Crashed tests:
	
[ERROR] org.apache.wayang.tensorflow.model.TensorflowModelTest
	
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:643)
	
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
	
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:250)
	
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1203)
	
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1055)
	
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:871)
	
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:126)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:328)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:316)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:174)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.access$000(MojoExecutor.java:75)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:162)
	
[ERROR] 	at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:159)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:105)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:73)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:53)
	
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:118)
	
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:261)
	
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:173)
	
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:101)
	
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:906)
	
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:283)
	
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:206)
	
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
	
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
	
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
	
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
	
[ERROR] 
	
[ERROR] -> [Help 1]
	
[ERROR] 
	
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
	
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
	
[ERROR] 
	
[ERROR] For more information about the errors and possible solutions, please read the following articles:
	
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
	
[ERROR] 
	
[ERROR] After correcting the problems, you can resume the build with the command
	
[ERROR]   mvn <args> -rf :wayang-tensorflow
```

Similar issue we had before not sure if this has been fixed for some other test.

--- a/wayang-platforms/wayang-tensorflow/src/test/java/org/apache/wayang/tensorflow/model/TensorflowModelTest.java
+++ b/wayang-platforms/wayang-tensorflow/src/test/java/org/apache/wayang/tensorflow/model/TensorflowModelTest.java
@@ -71,15 +71,22 @@ class TensorflowModelTest {
         try (TensorflowModel tfModel = new TensorflowModel(model, criterion, optimizer, acc)) {
             System.out.println(tfModel.getOut().getName());
             tfModel.train(x, y, 100, 6);
-            TFloat32 predicted = tfModel.predict(x);
-            Ops tf = Ops.create();
-            org.tensorflow.op.math.ArgMax<TInt32> argMax = tf.math.argMax(tf.constantOf(predicted), tf.constant(1), TInt32.class);
-            final TInt32 tensor = argMax.asTensor();
-            System.out.print("[ ");
-            for (int i = 0; i < tensor.shape().size(0); i++) {
-                System.out.print(tensor.getInt(i) + " ");
+            try (TFloat32 predicted = tfModel.predict(x)) {
+                System.out.print("[ ");
+                for (int i = 0; i < predicted.shape().size(0); i++) {
+                    float max = -Float.MAX_VALUE;
+                    int maxIdx = -1;
+                    for (int j = 0; j < predicted.shape().size(1); j++) {
+                        float val = predicted.getFloat(i, j);
+                        if (val > max) {
+                            max = val;
+                            maxIdx = j;
+                        }
+                    }
+                    System.out.print(maxIdx + " ");
+                }
+                System.out.println("]");
             }
-            System.out.println("]");
         }
         System.out.println();
     }

